### DISCUSSION TITLE
Inline Component Pagination - Computed Properties

### DISCUSSION THREAD

**Discussion by Infamoustrey** [livewire/livewire#5977] [2. Show and tell]
Not sure if this was obvious to others but I thought I'd share a little hiccup that @calebporzio helped me nab. If you have an inline component but want to add pagination to it you can use a [computed property](https://livewire.laravel.com/docs/computed-properties#using-inline-templates) to pass the pagination object to the view. This seems obvious to me now but I had read through the docs a few times and it never clicked. Hopefully someone in the future searches and is able to find this. 

```php
<?php

namespace App\Livewire;

use Livewire\Attributes\Computed;
use Livewire\WithPagination;
use Livewire\Component;
use App\Models\Post;

class ShowPosts extends Component
{
    use WithPagination;

    #[Computed()]
    public function posts()
    { 
        return Post::paginate(10);
    }

    public function render()
    { 
        return <<<'HTML'
            <div>
                <div>
                    @foreach ($this->posts as $post)
                        <!-- ... -->
                    @endforeach
                </div>

                {{ $this->posts->links() }}
            </div>
        HTML; 
    }
}
```

Hope this helps! I'm loving V3 and love the helping nature of everyone else here!

**Comment by joshhanley** [comment]
@Infamoustrey thanks!

**Comment by JulioSimon** [comment]
I was building a category page today and thought about the same ^^. It's awesome to have the pagination and filters in a computed property. I wonder if the computed properties cache also works with the pagination trait, does anyone tested it?

**Reply by Life-Gate-Mission-Hospital** [reply]
It doesn't with cached or persist
